allow supported resource overrides in the limitranger plugin

pweil- opened the pull request:
@derekwaynecarr - found when implementing limit ranges for image resources post rebase.  PTAL

Commit:
allow supported resource overrides in the limitranger plugin

derekwaynecarr commented:
@pweil- can we combine `LimitFunc` and `SupportedResourcesFunc` into a common interface?  They basically need to be specified in lock-step so having them be provided separately feels error prone.

derekwaynecarr commented:
@pweil- I am also fine if we defer that to a future PR, but put a TODO to merge the two into a common interface.

pweil- commented:
Per discussion, updated to implement as an interface.  Also added the ability to ignore limits if required with a default implementation to not ignore any limits.

derekwaynecarr commented on plugin/pkg/admission/limitranger/admission.go:
```diff
@@ -383,6 +375,38 @@ func sum(inputs []api.ResourceList) api.ResourceList {
+// SupportsAttributes ignores all calls that do not deal with pod resources since that is
+// all this supports now.  Also ignores any call that has a subresource defined.
+func (d *DefaultLimitRangerActions) SupportsAttributes(a admission.Attributes) bool {
+	if a.GetSubresource() != "" {
```
are we confident in our downstream use cases that we will not forget to include this line?

pweil- replied:
I can certainly move it back into the main admit method if you think it's worth it.  

Here were the two downstream cases I found:
https://github.com/openshift/origin/pull/7989/files#diff-5fd1547857aea1ef0cdb18cb293f88faR210
https://github.com/openshift/origin/pull/7989/files#diff-c0324af3f9610f91b1fde15a63e35bf5R106

derekwaynecarr commented on plugin/pkg/admission/limitranger/interfaces.go:
```diff
@@ -17,9 +17,18 @@ limitations under the License.
+	// SupportsAttributes is a pluggable function to allow overridding what resources the limitranger
+	// supports.
+	SupportsAttributes(attr admission.Attributes) bool
+	// SupportsLimit is a pluggable function to allow ignoring limits that should not be applied
```
I assume this is when a limit is referencing a custom type?

pweil- replied:
Yes, used to filter limit ranges that only apply to images downstream: https://github.com/openshift/origin/pull/7989/files#diff-c0324af3f9610f91b1fde15a63e35bf5R118

derekwaynecarr commented:
LGTM

derekwaynecarr commented:
actually, looks good, but need to see what travis is upset about...

pweil- commented:
`No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`

reopening to kick
